### PR TITLE
Refactor adapterremoval

### PR DESCRIPTION
**Pull request by Midnighter**
## PR checklist

Insert the missing `.fastq` file extension so that other tools can recognize the file type.

- [x] This comment contains a description of changes (with reason).
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by Midnighter** [comment]
As per @jfy133 's request, rename the settings file from `.log` to `.settings`.

File: modules/adapterremoval/main.nf [hunk 1]
```diff
@@ -38,10 +37,25 @@ process ADAPTERREMOVAL {
             $adapterlist \\
             --basename ${prefix} \\
             --threads ${task.cpus} \\
-            --settings ${prefix}.log \\
+            --settings ${prefix}.settings \\
```
Comment (jfy133) [inline modules/adapterremoval/main.nf:40]:
```suggestion
```

Ok, I forgot that infact `$basename.settings` is default, so we can just remove it entirely.

File: modules/adapterremoval/main.nf [hunk 2]
```diff
@@ -12,15 +12,14 @@ process ADAPTERREMOVAL {
...
-    tuple val(meta), path("${prefix}.paired.gz")               , optional: true, emit: paired_interleaved
-    tuple val(meta), path('*.log')                             , emit: log
-    path "versions.yml"                                        , emit: versions
+    tuple val(meta), path("${prefix}.truncated.fastq.gz")            , optional: true, emit: singles_truncated
+    tuple val(meta), path("${prefix}.discarded.fastq.gz")            , optional: true, emit: discarded
+    tuple val(meta), path("${prefix}.pair{1,2}.truncated.fastq.gz")  , optional: true, emit: paired_truncated
+    tuple val(meta), path("${prefix}.collapsed.fastq.gz")            , optional: true, emit: collapsed
+    tuple val(meta), path("${prefix}.collapsed.truncated.fastq.gz")  , optional: true, emit: collapsed_truncated
+    tuple val(meta), path("${prefix}.paired.fastq.gz")               , optional: true, emit: paired_interleaved
+    tuple val(meta), path('*.settings')                              , emit: log
```
Comment (jfy133) [inline modules/adapterremoval/main.nf:21]:
```suggestion
    tuple val(meta), path('*.settings')                              , emit: settings
```

Comment (jfy133) [reply modules/adapterremoval/main.nf:21]: Would need to rename in `meta.yml`

File: modules/adapterremoval/main.nf [hunk 3]
```diff
@@ -38,10 +37,25 @@ process ADAPTERREMOVAL {
...
             --basename ${prefix} \\
             --threads ${task.cpus} \\
-            --settings ${prefix}.log \\
+            --settings ${prefix}.settings \\
             --seed 42 \\
             --gzip
 
+        ensure_fastq() {
+            if [ -f "\${1}" ]; then
+                mv "\${1}" "\${1::-3}.fastq.gz"
```
Comment (jfy133) [inline modules/adapterremoval/main.nf:46]: That's some nifty bash there ;)

File: modules/adapterremoval/main.nf [hunk 4]
```diff
@@ -38,10 +37,25 @@ process ADAPTERREMOVAL {
...
+
+        }
+
+        ensure_fastq '${prefix}.truncated.gz'
+        ensure_fastq '${prefix}.discarded.gz'
+        ensure_fastq '${prefix}.pair1.truncated.gz'
+        ensure_fastq '${prefix}.pair2.truncated.gz'
+        ensure_fastq '${prefix}.collapsed.gz'
+        ensure_fastq '${prefix}.collapsed.truncated.gz'
+        ensure_fastq '${prefix}.paired.gz'
```
Comment (jfy133) [inline modules/adapterremoval/main.nf:57]:
```suggestion
```

Shouldn't be present in single-end data, just `truncated` and `discarded`

**Review by jfy133 — APPROVED** [review]
Couple of polishing comments but otherwise LGTM :+1: thank you very much @Midnighter !

**Comment by jfy133** [comment]
I would normally say replace the `log` output channel to `settings` (I doubt anyone is really using this module other than pipeleines related to me anwyay), but not important :+1: good to merge!

**Pull request merged**